### PR TITLE
[EA] Add support of multi-lines error messages

### PR DESCRIPTION
**Pull request by Stargazer1998**
Current implementation is unable to provide a complete description for the errors composed of multiple lines. An example is as:

[2023-04-10 12:04:22.1045] Error - Error loading plugins from: C:\file1\idc\sample_ext.x64.dll
File name: 'file:///C:\AA\BB\System.Text.Json.dll' ---> System.NotSupportedException: An attempt was made to load an 
(moree lines!)
   at System.Reflection.Assembly.LoadFrom(String assemblyFile)
   at A.A`1.a(String , Predicate`1 , Object[] )
   at A.A`1.A(String , Predicate`1 , Object[] )
### Testing done
The current change was tested by adding additional test case to the EmbeddedEngineerParserTest file.
Tests were executed using [warnings-ng-plugin-devenv] development environment.
Screenshots of the test results:
test_1
test_2

### Submitter checklist
- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [X] Link to relevant pull requests, esp. upstream and downstream changes
- [X] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by uhafner** [comment]
Can you please have a look at the style checks?

**Comment by Stargazer1998** [comment]
> Can you please have a look at the style checks?



> Can you please have a look at the style checks?

It is now done.

File: src/main/java/edu/hm/hafner/analysis/parser/EmbeddedEngineerParser.java [hunk 1]
```diff
@@ -40,28 +45,47 @@ public Report parse(final ReaderFactory reader) throws ParsingException {
...
+        List<String> logInfo = new ArrayList<>();
+
+        while (lineIterator.hasNext()) {
+            String currentLine = lineIterator.next();
+            Matcher logBeginningMatcher = LOG_BEGINNING_PATTERN.matcher(currentLine);
+            if (logBeginningMatcher.matches()) {
+                logInfo.add(currentLine);
+            }
+            else {
+                int lastIdx = logInfo.size() - 1;
```
Comment (uhafner) [inline src/main/java/edu/hm/hafner/analysis/parser/EmbeddedEngineerParser.java:58]: What happens if `logInfo` is empty here? Then the next statements need to be skipped.

File: src/main/java/edu/hm/hafner/analysis/parser/EmbeddedEngineerParser.java [hunk 2]
```diff
@@ -40,28 +45,47 @@ public Report parse(final ReaderFactory reader) throws ParsingException {
...
+                StringBuilder multiLineInfo = new StringBuilder(logInfo.get(lastIdx));
+                multiLineInfo.append(" ").append(currentLine);
+                logInfo.set(lastIdx, multiLineInfo.toString());
+            }
+        }
+        return logInfo;
```
Comment (uhafner) [inline src/main/java/edu/hm/hafner/analysis/parser/EmbeddedEngineerParser.java:64]: I'm not sure how big your logfiles are getting, but you are basically defeating the purpose of the stream by reading everything into memory. Other parsers that want to join additional lines use `LookaheadStream`. See GccParser or MavenParser as an example.

**Review by uhafner — CHANGES_REQUESTED** [review]

File: src/main/java/edu/hm/hafner/analysis/parser/EmbeddedEngineerParser.java (hunk 2 above)
Comment (Stargazer1998) [reply src/main/java/edu/hm/hafner/analysis/parser/EmbeddedEngineerParser.java]: thank you for the suggestion.
implementation has been changed to use `LookaheadStream`.

**Review by uhafner — APPROVED** [review]

**Comment by uhafner** [comment]
Thanks! Can you please merge with main (or rebase)?

**Comment by Stargazer1998** [comment]
> Thanks! Can you please merge with main (or rebase)?

Everything now seems to be okay! Thank you for all the updates.

**Review by uhafner — APPROVED** [review]

**Pull request merged**